### PR TITLE
Modify rule S4428: LaYC format

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**

### CODE REVIEW

File: rules/S4428/description.adoc [hunk 1]
```diff
@@ -1,4 +1,6 @@
-The ``++PartCreationPolicyAttribute++`` attribute, which is part of the Managed Extensibility Framework (MEF), is used to specify how the exported object will be created. Therefore it doesn't make sense not to export this a class with this attribute using the ``++ExportAttribute++`` attribute.
+== Why is this an issue?
 
+In https://learn.microsoft.com/en-us/dotnet/framework/mef/[Managed Extensibility Framework] (MEF), if you want to modify the default behavior for an export then you need to use the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[`PartCreationPolicyAttribute`] attribute.
+However, as we are talking about an export, you should not forget to also annotate your class with the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.exportattribute[`ExportAttribute`] othewise the `PartCreationPolicyAttribute` has no meaning.
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/description.adoc:4]:
```suggestion
For the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[`PartCreationPolicyAttribute`] to be meaningful in the context of an export, it is necessary to also annotate the class with the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.exportattribute[`ExportAttribute`].
```

File: rules/S4428/resources.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+== Resources
+
+=== Documentation
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/framework/mef/[Managed Extensibility * Framework (MEF)]
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/resources.adoc:4]:
```suggestion
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/framework/mef/[Managed Extensibility Framework (MEF)]
```

File: rules/S4428/resources.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/framework/mef/attributed-programming-model-overview-mef[Attributed programming model overview (MEF)]
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[PartCreationPolicyAttribute Class]
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.exportattribute[ExportAttribute Class]
+
+=== Articles & blog posts
+
+* Stefan Henneken - https://stefanhenneken.net/2015/11/08/mef-part-1-fundamentals-imports-and-exports/[MEF Part 1 – Fundamentals, Imports and Exports]
+* Stefan Henneken - https://stefanhenneken.net/2019/01/26/mef-part-2-metadata-and-creation-policies/[MEF Part 2 – Metadata and creation policies]
+* Stefan Henneken - https://stefanhenneken.net/2019/03/06/mef-part-3-life-cycle-management-and-monitoring/[MEF Part 3 – Life cycle management and monitoring]
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/resources.adoc:13]: The linked pages look broken in Island browser (Missing images, code snippets are misaligned). Did you check with other browsers?

File: rules/S4428/resources.adoc [hunk 4]
```diff
@@ -0,0 +1,13 @@
+== Resources
+
+=== Documentation
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/framework/mef/[Managed Extensibility * Framework (MEF)]
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/framework/mef/attributed-programming-model-overview-mef[Attributed programming model overview (MEF)]
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[PartCreationPolicyAttribute Class]
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/resources.adoc:6]: Consider adding a link to https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.creationpolicy

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
Some formatting issues and I got some help from my friend GPT with the wording.

File: rules/S4428/resources.adoc (hunk 3 above)
Comment (mary-georgiou-sonarsource) [reply rules/S4428/resources.adoc:13]: Yes - they work with Chrome normally - it's island the issue.
Is this a reason to remove them? (not showing well in all browsers?)
I found them pretty useful.

File: rules/S4428/description.adoc [hunk 5]
```diff
@@ -1,4 +1,6 @@
-The ``++PartCreationPolicyAttribute++`` attribute, which is part of the Managed Extensibility Framework (MEF), is used to specify how the exported object will be created. Therefore it doesn't make sense not to export this a class with this attribute using the ``++ExportAttribute++`` attribute.
+== Why is this an issue?
 
+To customize the default behavior for an export in the https://learn.microsoft.com/en-us/dotnet/framework/mef/[Managed Extensibility Framework] (MEF), applying the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[`PartCreationPolicyAttribute`] is necessary.
+For the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[`PartCreationPolicyAttribute`] to be meaningful in the context of an export, it is necessary to also annotate the class with the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.exportattribute[`ExportAttribute`].
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/description.adoc:4]: necessary is repeated twice. I would suggest using a synonym.
```suggestion
For the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute[`PartCreationPolicyAttribute`] to be meaningful in the context of an export, it is required to also annotate the class with the https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.exportattribute[`ExportAttribute`].
```

File: rules/S4428/csharp/rule.adoc [hunk 6]
```diff
@@ -1,29 +1,28 @@
...
+== How to fix it
+
+=== Code examples
+
+==== Noncompliant code example
 
-[source,csharp]
+[source,csharp,diff-id=1,diff-type=noncompliant]
 ----
 [PartCreationPolicy(CreationPolicy.Any)] // Noncompliant
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/csharp/rule.adoc:11]: Consider adding a `using System.ComponentModel.Composition;`

File: rules/S4428/vbnet/rule.adoc [hunk 7]
```diff
@@ -1,20 +1,22 @@
...
+== How to fix it
+
+=== Code examples
 
-[source,vbnet]
+==== Noncompliant code example
+
+[source,vbnet,diff-id=1,diff-type=noncompliant]
 ----
 <PartCreationPolicy(CreationPolicy.Any)> ' Noncompliant
```
Comment (martin-strecker-sonarsource) [inline rules/S4428/vbnet/rule.adoc:11]: Same here: consider adding an `Imports System.ComponentModel.Composition`

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. Just some optional fine-tuning suggestions.

**Pull request merged**